### ISSUE TITLE
add_shared_layer conflicts with batch-norm?

### ISSUE THREAD

**Issue by lxmeng** [fchollet/keras#1740]
i construct a siamese networks for text matching, when i add a batch-norm layer after conv-layer, i found that acc decreases heavily? anyone meets this problems or can anybody gives a explanation?

**Comment by neggert** [comment]
Are you setting the normalization axis correctly? This just got me. It defaults to normalizing over the last axis, which if you're using images with the Theano channel convention, isn't what you want.
